TOOLSDOC-645: Maven basics article

mishaone opened the pull request:
@xcoulon - Added an article about basic Maven configuration. This has been reviewed for technical accuracy by rawagner as the QE Maven expert, but if you have any further feedback, please let me know.

maxandersen commented:
@fbricon could you spend some eye time on this one to review ?

maxandersen commented on documentation/howto/maven_basics.adoc:
```diff
@@ -0,0 +1,147 @@
+:experimental:
+:imagesdir: ./images
+
+In the context of application development, Maven allows you to extend your applications by referencing and utilizing resources from local and remote repositories. This article serves as an introduction to using Maven with the IDE and contains instructions for the following tasks:
```
I think using the word "extend" for what Maven does is unfortunate. It is not 100% incorrect but it is a misleading term imo.

Maven is much more about providing standardized build system for your projects, and *one* part of this build system is that it allows you to fetch dependencies from one or multiple repositories.

mishaone replied:
Thanks for the clarification. I've updated the text to more accurately reflect the benefit of using Maven.

maxandersen commented on documentation/howto/maven_basics.adoc:
```diff
@@ -0,0 +1,147 @@
+. <<New_Module,Create a New Maven Module>>
+. <<Maven_Support,Add Maven Support to an Existing Non-Maven Project>>
+
+Maven projects serve as parents for Maven modules. For each Maven module attached to a Maven project, a section is added to the project’s pom.xml file. Each command run by the Maven project is run against each of the included modules first.
```
"parent", "attached" and "command" are not the correct terms in this case. And no, each command is not always run against each of the included modules first.

My suggestion: 
"Root Maven projects can serve as aggregators for multiple Maven modules. For each module that is part of a maven project, a `<module>` entry is added to the project's `pom.xml` file. A `pom.xml` that contains `<module>` entries are often referred to as an aggregator pom.

When modules are included into a project it is possible to execute Maven Goals across all of the modules by a single command."

mishaone replied:
Sounds good, replaced text with proposed version.

maxandersen commented on documentation/howto/maven_basics.adoc:
```diff
@@ -0,0 +1,147 @@
+.. Type `maven` in the `Filter` field and select `Maven Project`.
+.. Click `Next` to continue.
+. In the first step of the `New Maven Project` wizard, enter the initial project details:
+.. Select the `Use default Workspace location` check box to use the default workspace. If required, specify a non-default location by deselecting the check box and using the `Browse` button to specify the desired target directory.
```
we should recommend users to *not* use default workspace location. it is much easier to reuse your projects with other tools (such as the command line) if they do not live inside eclipse workspaces.

mishaone replied:
Done. New text: "Ensure that the Use default Workspace location check box is not selected and specify a non-default location for your workspace files using the Browse button. Using a non-default workspace location is recommended because this allows other tools to access the workspace location easily."

maxandersen commented on documentation/howto/maven_basics.adoc:
```diff
@@ -0,0 +1,147 @@
+[[New_Module]]
+== 2. Create a New Maven Module
+
+Each Maven project can include multiple independent Maven modules. Follow the instructions to create your first Maven module:
```
some of these modules could also be dependent on each other so I would remove the `independent` part here.

mishaone replied:
Removed "independent".

maxandersen commented on documentation/howto/maven_basics.adoc:
```diff
@@ -0,0 +1,147 @@
+Each Maven project can include multiple independent Maven modules. Follow the instructions to create your first Maven module:
+
+=== Prerequisites
+. You must have an existing Maven project available to use for the new module. See <<New_Project,Create a New Maven Project>> for instructions to create a new Maven project.
```
is this really necessary - its about 10 seconds ago I read that on the same page?

mishaone replied:
It may seem a bit redundant, but this is mostly in case someone lands on this particular part of the document because they want to create a module, but possibly don't know that a project is a prerequisite. The general docs approach to this sort of thing is better to have clear instructions about prerequisites because users rarely read a document from start to finish and usually skip ahead to the specific procedure they want. If this approach is not desired for JBT, I'm happy to remove this bit, though.

maxandersen commented on documentation/howto/maven_basics.adoc:
```diff
@@ -0,0 +1,147 @@
+.. Click `Next` to continue.
+. In the third step of the `New Maven Module` wizard, enter the archetype details:
+.. Add a unique group ID to the `Group Id` field. Note that this ID must be different from the group ID of the included Maven project.
+.. Add the desired version number in the `Version` field. As an example, the version `0.0.1-SNAPSHOT` was used as the version number for the previously created Maven project, therefore here we can use a variation such as `0.1.1-SNAPSHOT`.
```
not following why the previous created project version is presented as affecting the version number of this new one ?

mishaone replied:
Changed this to:

"Add the desired version number in the Version field. For details about the appropriate version build numbers, see Project Versions[link]"

maxandersen commented on documentation/howto/maven_basics.adoc:
```diff
@@ -0,0 +1,147 @@
+The previous tasks contain instructions to create a new Maven project and Maven module. However, for an existing application that was not created with Maven support, use the following instructions to add Maven support to the non-Maven project:
+
+==== Adding Maven Support to an Existing Non-Maven Project ====
+. (Optional) To add dependencies to the project, use the following instructions:
```
is there really a need to explain a user how to add dependencies to the project when the section is about converting projects already has dependencies ?

mishaone replied:
This was added based on feedback from @rawagner. As I understand it, it's an optional mini-task that users may seek to do here instead of or in addition to changing existing dependencies. Happy to take it out if this is unnecessary after all.

fbricon replied:
agree with max, that should not be part of that doc.

maxandersen commented on documentation/howto/maven_basics.adoc:
```diff
@@ -0,0 +1,147 @@
+image::mvnbasic_create_new_pom.png[Create a New Pom Descriptor]
++
+.. Click `Finish` to finalize the pom information.
+. If dependencies were added (using the instructions in step 1), a wizard appears displaying all added dependencies and a green check mark when each dependency is identified.
```
should it not explain how this identification is made and what to do in the case dependencies are not matched?

mishaone replied:
Good point. I'll research and propose something for the "Do You Know" section about how the identification happens and experiment with possible errors and how to resolve them for the troubleshooting section.

maxandersen commented on documentation/howto/maven_basics.adoc:
```diff
@@ -0,0 +1,147 @@
+. In the third step of the `New Maven Project` wizard, specify the archetype parameters:
+.. Enter the desired group ID in the `Group Id` field.
+.. Enter the desired artifact ID in the `Artifact Id` field.
+.. Set the `Version` field’s value to `0.0.1-SNAPSHOT` (or any other unique value).
```
no, it cannot just be any unique value.

Maven has rules for how a version value looks like and there are also recommendations.

Plus, -SNAPSHOT has a very specific meaning. 

see details at http://books.sonatype.com/mvnref-book/reference/pom-relationships-sect-pom-syntax.html

mishaone replied:
Added explanation and link to build version information to both instances (in second, removed recommendation to add a variant of the SNAPSHOT value).

maxandersen commented on documentation/howto/maven_basics.adoc:
```diff
@@ -0,0 +1,147 @@
+
+In the context of application development, Maven allows you to extend your applications by referencing and utilizing resources from local and remote repositories. This article serves as an introduction to using Maven with the IDE and contains instructions for the following tasks:
+
+. <<New_Project,Create a New Maven Project>>
```
if you add a

:toc 
toc::[]

you get a toc automatically.

mishaone replied:
I didn't realize there was an automatic ToC available. At a quick look, I can't find anything beyond instructions to add :toc: to the definitions at the beginning of the document and that doesn't seem to quite do the trick by itself. I'll figure this out and add it for future docs. I used the recommended method in the z_howto template here since it appeared to be the recommended way.

mishaone commented:
Update: I've implemented most of Max's feedback. The last item (how dependency identification happens and troubleshooting common errors), which is a bit of a minor RFE I am looking into will be added tomorrow once I draft this content and request feedback.

If there is any further feedback, please do comment (Max, or Fred if he has the time to have a look tomorrow).

mishaone commented:
Updated with a Do You Know section and a Troubleshooting section as requested. @maxandersen and @fbricon, could you have a look today if at all possible and let me know if there is any further feedback please?

fbricon commented on documentation/howto/index.adoc:
```diff
@@ -7,6 +7,8 @@ The How To section contains articles describing features of JBoss Tools and Deve
 
```
previous line : typo on specfic tasks

mishaone replied:
Good catch. This is now fixed.

fbricon commented on documentation/howto/maven_basics.adoc:
```diff
@@ -0,0 +1,183 @@
+. <<New_Module,Create a New Maven Module>>
+. <<Maven_Support,Add Maven Support to an Existing Non-Maven Project>>
+
+Root Maven projects can serve as aggregators for multiple Maven modules. For each module that is part of a maven project, a <module> entry is added to the project's pom.xml file. A pom.xml that contains <module> entries are often referred to as an aggregator pom.
```
for multiple Maven modules, or sub-projects.

mishaone replied:
Fixed line.

fbricon commented on documentation/howto/maven_basics.adoc:
```diff
@@ -0,0 +1,183 @@
+
+Root Maven projects can serve as aggregators for multiple Maven modules. For each module that is part of a maven project, a <module> entry is added to the project's pom.xml file. A pom.xml that contains <module> entries are often referred to as an aggregator pom.
+
+When modules are included into a project it is possible to execute Maven goals across all of the modules by a single command.
```
single command issued from the parent project directory

mishaone replied:
Added to line.

fbricon commented on documentation/howto/maven_basics.adoc:
```diff
@@ -0,0 +1,183 @@
+:imagesdir: ./images
+:toc:
+
+In the context of application development, Maven provides a standardized build system for projects. One of the main benefits of using Maven with your project is that it facilitates fetching dependencies from one or more repositories. This article serves as an introduction to using Maven with the IDE and contains instructions for the following tasks:
```
introduction to using Maven with the IDE : seems a bit too generic. These instructions are for Eclipse-base IDEs

mishaone replied:
I'm afraid I don't quite know the difference between the functionality offered by Maven for Eclipse-based IDEs vs general IDEs. Any chance you can help clarify this, @fbricon? This introduction is from the 7.1 docs for Maven and seemed valid still to me.

fbricon replied:
I agree the general concepts (modules, dep. mgmt) are not specific to Eclipse based IDE, but all the instructions hereafter are really about using Eclipse wizards and features

fbricon commented on documentation/howto/maven_basics.adoc:
```diff
@@ -0,0 +1,183 @@
+
+[[New_Project]]
+== 1. Create a New Maven Project
+The first step when using Maven with an IDE is to create a Maven project that can contain multiple independent Maven modules. Follow the instructions to create a Maven project:
```
with an Eclipse-based IDE ?

mishaone replied:
Changed according to suggestion since the introduction is apparently Eclipse-based IDE-specific.

fbricon commented on documentation/howto/maven_basics.adoc:
```diff
@@ -0,0 +1,183 @@
+.. Ensure that the `Use default Workspace location` check box is not selected and specify a non-default location for your workspace files using the `Browse` button. Using a non-default workspace location is recommended because this allows other tools to access the workspace location easily.
+.. The following optional steps are available for further configuration, if required:
+... (Optional) Select the `Create a simple project (skip archetype selection)` check box. If this check box is selected, the `Select an Archetype` step of the wizard is skipped.
+... (Optional) Select the `Add project(s) to working set` check box to add additional projects to the working set.
```
to add the newly created projects to a working set

mishaone replied:
Implemented change.

fbricon commented on documentation/howto/maven_basics.adoc:
```diff
@@ -0,0 +1,183 @@
+.. The following optional steps are available for further configuration, if required:
+... (Optional) Select the `Create a simple project (skip archetype selection)` check box. If this check box is selected, the `Select an Archetype` step of the wizard is skipped.
+... (Optional) Select the `Add project(s) to working set` check box to add additional projects to the working set.
+... (Optional) Click `Advanced` to view additional optional advanced configuration for the new Maven project, such as whether to `Resolve Workspace projects`, `Profiles`, and any `Name templates` to set a desired naming convention for the project.
```
* `Resolve Workspace projects` : dependencies opened as workspace projects will be resolved without having to install them to your local Maven repository first. This way, any change made to one of these dependencies will have an immediate effect on other projects consuming it (compilation, refactoring...). When workspace resolution is disabled, dependencies existing in the workspace must be installed to your local Maven repository after any change (by running `mvn install`), in order to see effects in projects consuming them.
* `Profiles` : select a set of Maven profiles to activate or deactivate in the workspace. Profiles are defined in the project pom.xml, or inherited from a parent pom.xml, or defined in the use settings.xml.
* `Name templates` : allows you to disambiguate projects names in the workspace by prepending or appending group Id or SCM branch names to the default artifact Id

mishaone replied:
Thanks, added information to both instances for these fields.

fbricon commented on documentation/howto/maven_basics.adoc:
```diff
@@ -0,0 +1,183 @@
++
+.. Click `Next` to continue.
+. In the third step of the `New Maven Project` wizard, specify the archetype parameters:
+.. Enter the desired group ID in the `Group Id` field.
```
A Group Id is akin to your organization namespace, eg. com.company.businessunit.project

mishaone replied:
Added this in each instance where group ID is to be configured.

fbricon commented on documentation/howto/maven_basics.adoc:
```diff
@@ -0,0 +1,183 @@
+.. Click `Next` to continue.
+. In the third step of the `New Maven Project` wizard, specify the archetype parameters:
+.. Enter the desired group ID in the `Group Id` field.
+.. Enter the desired artifact ID in the `Artifact Id` field.
```
The artifact Id is the actual name of your project. It can not contain spaces. The only special characters allowed are : `.`, `_`, and `-`

mishaone replied:
Added to the one mention of adding an artifact ID.

fbricon commented on documentation/howto/maven_basics.adoc:
```diff
@@ -0,0 +1,183 @@
+[[New_Module]]
+== 2. Create a New Maven Module
+
+Each Maven project can include multiple Maven modules. Follow the instructions to create your first Maven module:
```
Each Maven project with a packaging `pom`

mishaone replied:
Fixed.

fbricon commented on documentation/howto/maven_basics.adoc:
```diff
@@ -0,0 +1,183 @@
+.. In the `Project Explorer` view, expand your existing Maven project and double-click `pom.xml` in the expanded view.
+.. The `Overview` tab appears with configuration information for your Maven project.
+.. In the `Packaging` drop-down menu, select the `pom` option.
++
```
Misha, changing an existing project packaging to `pom` is not recommended. For instance, if you create a `jar` project and later change its pakaging to `pom`, it would still be able to compile classes in the IDE, because the Java nature would not have been removed. But CLI builds would not compile anything.

I would rather recommend to create a simple Maven project of type `pom`, by skipping the archetype selection

mishaone replied:
Thanks for clarifying this, Fred. I've updated the optional step of the first procedure to recommend people check the skip archetype selection option and removed this recommendation from the prerequisites.

fbricon commented on documentation/howto/maven_basics.adoc:
```diff
@@ -0,0 +1,183 @@
+
+=== Creating a New Maven Module
+. Create a new Maven module as follows:
+.. In your workspace, navigate to menu:File[New > Other].
```
No, rather, right-click on a pom project in your workspace, then [New > Project]

mishaone replied:
@fbricon, could you elaborate in this please? I assumed since we have separate options for "Maven Project" and "Maven Module" under Other, these are the two used for a new Maven project and module respectively. Would project instead here just create another project rather than a module?

fbricon replied:
ok I wasn't clear. The proper way to do it is : right-click on a pom project in your workspace, then [New > Project] _then_  From the wizard selection screen, expand `Maven` and select `Maven Module`.

fbricon commented on documentation/howto/maven_basics.adoc:
```diff
@@ -0,0 +1,183 @@
+.. From the wizard selection screen, expand `Maven` and select `Maven Module`.
+.. Click `Next >` to continue.
+. In the first step of the `New Maven Module` wizard, enter the initial module details:
+.. Enter the desired module name in the `Module Name` field.
```
That will correspond to your project artifactId

mishaone replied:
Updated.

fbricon commented on documentation/howto/maven_basics.adoc:
```diff
@@ -0,0 +1,183 @@
+.. Click `Next >` to continue.
+. In the first step of the `New Maven Module` wizard, enter the initial module details:
+.. Enter the desired module name in the `Module Name` field.
+.. Click `Browse` from the `Parent Project` field to select an existing Maven project from the displayed list, then click `OK`.
```
Not necessary if you already started from an existing pom project

mishaone replied:
Removed.

fbricon commented on documentation/howto/maven_basics.adoc:
```diff
@@ -0,0 +1,183 @@
+.. Click `Browse` from the `Parent Project` field to select an existing Maven project from the displayed list, then click `OK`.
+.. The following optional steps are available for further configuration, if required:
+... (Optional) Select the `Create a simple project (skip archetype selection)` check box. If this check box is selected, the `Select an Archetype` step of the wizard is skipped.
+... (Optional) Select the `Add project(s) to working set` check box to add additional projects to the working set.
```
see previous comment on working sets

mishaone replied:
Fixed.

fbricon commented on documentation/howto/maven_basics.adoc:
```diff
@@ -0,0 +1,183 @@
+.. The following optional steps are available for further configuration, if required:
+... (Optional) Select the `Create a simple project (skip archetype selection)` check box. If this check box is selected, the `Select an Archetype` step of the wizard is skipped.
+... (Optional) Select the `Add project(s) to working set` check box to add additional projects to the working set.
+... (Optional) Click `Advanced` to view additional optional advanced configuration for the new Maven project, such as whether to `Resolve Workspace projects`, `Profiles`, and any `Name templates` to set a desired naming convention for the project.
```
see previous comment on advanced configuration

mishaone replied:
Fixed.

fbricon commented on documentation/howto/maven_basics.adoc:
```diff
@@ -0,0 +1,183 @@
++
+.. Click `Next` to continue.
+. In the third step of the `New Maven Module` wizard, enter the archetype details:
+.. Add a unique group ID to the `Group Id` field. Note that this ID must be different from the group ID of the included Maven project.
```
NO! It's customary to use the same groupId as the parent pom, for modules

mishaone replied:
Fixed.

fbricon commented on documentation/howto/maven_basics.adoc:
```diff
@@ -0,0 +1,183 @@
++
+
+**Result:** Your new Maven module is created and appears in the `Project Explorer` view.
+
```
Note that you can have a hierarchical view of the nested projects in the project Explorer view (see https://www.eclipse.org/eclipse/news/4.5/M5/#nested-projects)

mishaone replied:
Added information and link to details for the feature.

fbricon commented on documentation/howto/maven_basics.adoc:
```diff
@@ -0,0 +1,183 @@
++
+.Identify Maven Dependencies
+image::mvnbasic_convert_dependencies.png[Identify Maven Dependencies]
++
```
Click on "delete original references from project" to avoid keeping duplicate or stale dependencies in your project

fbricon replied:
All unchecked dependencies will be ignored and not be added to the generated pom.xml. However some of these can end up being added as transitive dependencies to your project. For instance, if you add `jsp-api` but remove `servlet-api`, the latter will still show up in the project classpath, as it's a dependency of `jsp-api`

fbricon replied:
You can double-click on a dependency (or click the edit button) to edit it's maven coordinates or scope.

Selecting several dependencies (ctrl+click) and clicking on the edit button will allow you to batch edit their scope. For instance set the `provided` scope to all dependencies that we know will be provided by an application server.

mishaone replied:
Added the first item as an additional step and the rest in Did You Know as that level of information may not be readily applicable to the task, but is useful to know.

fbricon commented on documentation/howto/maven_basics.adoc:
```diff
@@ -0,0 +1,182 @@
+. <<New_Module,Create a New Maven Module>>
+. <<Maven_Support,Add Maven Support to an Existing Non-Maven Project>>
+
+Root Maven projects can serve as aggregators for multiple Maven module, also known as sub-projects. For each module that is part of a maven project, a <module> entry is added to the project's pom.xml file. A pom.xml that contains <module> entries are often referred to as an aggregator pom.
```
multiple Maven module*s*

mishaone replied:
Fixed typo.

fbricon commented on documentation/howto/maven_basics.adoc:
```diff
@@ -0,0 +1,182 @@
+
+[[New_Project]]
+== 1. Create a New Maven Project
+The first step when using Maven with an Eclipse-based IDE is to create a Maven project that can contain multiple independent Maven modules. Follow the instructions to create a Maven project:
```
I'm confused here. If you want to explain how to create a Maven project that can contain multiple independent Maven modules, then you should not go on and describe how to use archetypes (which very rarely create pom projects)

mishaone replied:
Removed all mentions of archetypes and slightly rewrote the new project and new module procedures as required.

fbricon commented on documentation/howto/maven_basics.adoc:
```diff
@@ -0,0 +1,182 @@
+image::mvnbasic_project_name_location.png[Create a New Maven Project]
++
+.. When the configuration is complete, click `Next` to continue.
+. In the second step of the `New Maven Project` wizard, enter the archetype details:
```
At this point, you're supposed to create a simple maven projects, so archetype instructions are irrelevant here. They should be moved to another use-case

mishaone replied:
Done, sorry about that. The archetype step is excluded and the procedure amended accordingly for both the tasks.

mishaone commented:
@fbricon I've fixed up the two procedures and the contents should now be ready to go. Please let me know if you have any further feedback or if you ACK the content as technically correct.

fbricon commented:
There are still some problems. I'll comment in detail soon(tm)

fbricon commented on documentation/howto/maven_basics.adoc:
```diff
@@ -0,0 +1,182 @@
+[[New_Project]]
+== 1. Create a New Maven Project
+The first step when using Maven with an Eclipse-based IDE is to create a Maven project that can contain multiple independent Maven modules. Follow the instructions to create a Maven project:
+
```
This first sentence means you **have** to create multi-module projects. This is not necessarily true. You can very well create a project without a parent pom and without it being a module

mishaone replied:
Amended statement

fbricon commented on documentation/howto/maven_basics.adoc:
```diff
@@ -0,0 +1,182 @@
+.. Click `Next` to continue.
+. In the first step of the `New Maven Project` wizard, enter the initial project details:
+.. Ensure that the `Use default Workspace location` check box is not selected and specify a non-default location for your workspace files using the `Browse` button. Using a non-default workspace location is recommended because this allows other tools to access the workspace location easily.
+.. Select the `Create a simple project (skip archetype selection)` check box. If this check box is selected, the `Select an Archetype` step of the wizard is skipped and the project type is set to *pom*, which is required to create a Maven Module based on this Maven project.
```
s/Select/Check

mishaone replied:
s/select/check for all check box instructions.

fbricon commented on documentation/howto/maven_basics.adoc:
```diff
@@ -0,0 +1,172 @@
+
+=== Creating a New Maven Module
+. Create a new Maven module as follows:
+.. In your workspace, navigate to menu:File[New > Other].
```
In the project explorer view, right click on your recently created pom project and select New > Project.

mishaone replied:
Fixed first line of instructions.

fbricon commented on documentation/howto/maven_basics.adoc:
```diff
@@ -0,0 +1,172 @@
+.. From the wizard selection screen, expand `Maven` and select `Maven Module`.
+.. Click `Next >` to continue.
+. In the first step of the `New Maven Module` wizard, enter the initial module details:
+.. Select the `Create a simple project (skip archetype selection)` check box. If this check box is selected, the `Select an Archetype` step of the wizard is skipped.
```
You need to make sure Create a simple project is unchecked

mishaone replied:
Reverted changes.

fbricon commented on documentation/howto/maven_basics.adoc:
```diff
@@ -0,0 +1,172 @@
+.Set the Module Name and Parent
+image::mvnbasic_module_name_parent.png[Set the Module Name and Parent]
++
+.. When the configuration is complete, click `Next` to continue.
```
For this module, we want to describe how to use the archetype selection. We're not trying to create another pom project

mishaone replied:
Ack, changed this as part of the previous comment.

mishaone commented:
Thank you, @fbricon! I've implemented all the changes now and if you could have another look as soon as you can find the time, it would be much appreciated.

mishaone commented:
@fbricon - quick reminder if you could have a look at this please.

fbricon commented on documentation/howto/maven_basics.adoc:
```diff
@@ -0,0 +1,180 @@
+
+[[New_Project]]
+== 1. Create a New Maven Project
+The first step when using Maven with an Eclipse-based IDE is to create a Maven project. Maven projects may contain a single Maven module, multiple independent Maven modules or no Maven modules. Follow the instructions to create a Maven project:
```
Maven projects can be standalone, or comprised of multiple sub-projects, a.k.a modules

mishaone replied:
Thanks, Fred. Amended introduction to make this distinction clearer.

fbricon commented on documentation/howto/maven_basics.adoc:
```diff
@@ -0,0 +1,180 @@
+.. Type `maven` in the `Filter` field and select `Maven Project`.
+.. Click `Next` to continue.
+. Enter the initial project details:
+.. Check the `Create a simple project (skip archetype selection)` check box. If this check box is selected, the `Select an Archetype` step of the wizard is skipped and the project type is set to *pom*, which is required to create a Maven Module based on this Maven project.
```
Uncheck the box and follow the instructions in `Select a Module Archetype` if you want to create a standlone project.

mishaone replied:
Added.

mishaone commented:
@fbricon - updated according to feedback. Please have a look and comment.

fbricon commented on documentation/howto/maven_basics.adoc:
```diff
@@ -0,0 +1,180 @@
+.Select a Module Archetype
+image::mvnbasic_module_archetype.png[Select a Module Archetype]
++
+.. Click Next to continue.
```
beware, the following instructions pertain to the "create a simple project" path. You should just have to set/override the groupId, version, package and archetype properties if necessary

mishaone replied:
Added a note for users to be aware of this.

fbricon commented on documentation/howto/maven_basics.adoc:
```diff
@@ -0,0 +1,180 @@
+. Enter the module details:
+.. Add the same group ID value that was used for the Maven project to the `Group Id` field.
+.. Add the desired version number in the `Version` field. For details about the appropriate version build numbers, see http://books.sonatype.com/mvnref-book/reference/pom-relationships-sect-pom-syntax.html[Project Versions]
+.. Set the value of the `Packaging` field to `pom`.
```
no, we don't want to create a new pom project here

mishaone replied:
Fixed as part of replacing the screenshot and corrected instructions accordingly.

fbricon commented on documentation/howto/maven_basics.adoc:
```diff
@@ -0,0 +1,180 @@
+.. (Optional) Add a description for your module in the `Description` field.
++
+.Configure the Maven Module
+image::mvnbasic_configure_module.png[Configure the Maven Module]
```
Please use a screenshot of the Archetype properties page

mishaone replied:
Corrected screenshot and instructions.

fbricon commented:
Can you remove the part where you add jars to the non-maven project classpath?

mishaone commented:
Fixes:
* Corrected screenshot for module archetype parameters step
* Corrected instructions for the same step to correspond to the corrected screenshot
* Removed an optional step to add external jars to the classpath.

Question: Does this mean the troubleshooting errors that were due to adding incorrect or corrupt jar files in the external jar steps are now unnecessary? @fbricon could you comment on whether this information may still be valid for users or if I should just remove that as well?

mishaone commented:
@fbricon - if you have a moment, could you have a look at this?

fbricon commented on documentation/howto/maven_basics.adoc:
```diff
@@ -0,0 +1,180 @@
+image::mvnbasic_create_new_pom.png[Create a New Pom Descriptor]
++
+.. Click `Finish` to finalize the pom information.
+. If dependencies were added (using the instructions in step 1), a wizard appears displaying all added dependencies and a green check mark when each dependency is identified. Learn more about dependency identification in the <<did_you_know,Did You Know>> section.
```
There's no step1 instructions anymore. Rather "If the project references java dependencies, a wizard appears displaying all these dependencies ..."

mishaone replied:
Fixed.

fbricon commented on documentation/howto/maven_basics.adoc:
```diff
@@ -0,0 +1,180 @@
+. <<New_Module,Create a New Maven Module>>
+. <<Maven_Support,Add Maven Support to an Existing Non-Maven Project>>
+
+Root Maven projects can serve as aggregators for multiple Maven modules, also known as sub-projects. For each module that is part of a maven project, a <module> entry is added to the project's pom.xml file. A pom.xml that contains <module> entries are often referred to as an aggregator pom.
```
s/are often referred/is often referred/

mishaone replied:
Fixed.

mishaone commented:
@fbricon - could you have a look and merge it if there are no more changes required please?

fbricon commented:
so I made a few more adjustments, squashed the commits and merged it in master. See https://github.com/jbosstools/jbosstools-website/blob/master/documentation/howto/maven_basics.adoc

Thanks @mishaone !
